connectors/email/src/main/java/io/camunda/connector/email/client/jakarta/inbound/PollingManager.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.email.client.jakarta.inbound;
+
+import io.camunda.connector.api.inbound.InboundConnectorContext;
+import io.camunda.connector.email.client.jakarta.utils.Email;
+import io.camunda.connector.email.client.jakarta.utils.JakartaUtils;
+import io.camunda.connector.email.inbound.model.AllPollingConfig;
+import io.camunda.connector.email.inbound.model.EmailListenerConfig;
+import io.camunda.connector.email.inbound.model.PollingConfig;
+import io.camunda.connector.email.inbound.model.UnseenPollingConfig;
+import io.camunda.connector.email.response.ReadEmailResponse;
+import jakarta.mail.*;
+import jakarta.mail.search.FlagTerm;
+import java.util.Arrays;
+import org.eclipse.angus.mail.imap.IMAPMessage;
+
+public class PollingManager {
+
+  private final InboundConnectorContext connectorContext;
+  private final JakartaUtils jakartaUtils;
+  private final Folder folder;
+  private final Store store;
+
+  public PollingManager(
+      InboundConnectorContext connectorContext,
+      JakartaUtils jakartaUtils,
+      Folder folder,
+      Store store) {
+    this.connectorContext = connectorContext;
+    this.jakartaUtils = jakartaUtils;
+    this.folder = folder;
+    this.store = store;
+  }
+
+  public static PollingManager create(
+      InboundConnectorContext connectorContext, Folder folder, Store store) {
+    return new PollingManager(connectorContext, new JakartaUtils(), folder, store);
+  }
+
+  public void poll(EmailListenerConfig config) {
```

Comment:
I think `poll` should either return a `Collection<Message>` and then we deal with this result in a separate component.
Or maybe `poll(config, Consumer<Collection<Message>>)`.

Reply:
This is executed by a scheduler and therefore nothing returned from it will be taken into account, that makes me think exception should be managed better here

Reply:
And maybe we could use a `Callable` to deal with the result (I have no context I'll trust your judgement, just mentioning it)

Reply:
I kept the poll returning void as doing it like this would add a line and/or complexity within the scheduler Callable